addons/crm/models/crm_lead.py

```diff
@@ -1246,11 +1246,15 @@ def handle_salesmen_assignment(self, user_ids=None, team_id=False):
         :param int team_id: salesteam to assign
         """
         update_vals = {'team_id': team_id} if team_id else {}
-        for index, lead in enumerate(self):
-            if user_ids:
-                update_vals['user_id'] = user_ids[index % len(user_ids)]
-            if update_vals:
-                lead.write(update_vals)
+        if not user_ids:
+            self.write(update_vals)
+        else:
+            lead_ids = self.ids
+            steps = len(user_ids)
+            for idx in range(0, steps):
+                subset_ids = lead_ids[idx:len(lead_ids):steps]
```

Comment:
Maybe add a comment for dummies like me based on the method docstring:

```py
# pass 1 : lead_ids[0:6:4] = [L1,L5]
# pass 2 : lead_ids[1:6:4] = [L2,L6]
# pass 3 : lead_ids[2:6:4] = [L3]
# ...
```

But maybe it's overkill, I let you decide.